custom/enikshay/nikshay_datamigration/management/commands/create_enikshay_cases.py

```diff
@@ -0,0 +1,170 @@
+                    'phi': self.patient_detail.PHI,
+                    'first_name': self.patient_detail.first_name,
+                    'middle_name': self.patient_detail.middle_name,
+                    'last_name': self.patient_detail.last_name,
+                    'age': self.patient_detail.page,
+                    'sex': self.patient_detail.sex,
+                    'current_address': self.patient_detail.paddress,
+                    'mobile_number': self.patient_detail.pmob,
+                },
+            },
+        )
+
+    @memoized
+    def occurrence(self, outcome):
+        return CaseStructure(
+            attrs={
+                'create': True,
+                'case_type': 'occurrence',
+                'update': {
+                    'nikshay_id': outcome.PatientId.PregId,
+                    'hiv_status': outcome.HIVStatus,
+                },
+            },
+            indices=[CaseIndex(
+                self.person(),
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=self.person().attrs['case_type'],
+            )],
+        )
+
+    @memoized
+    def episode(self, outcome):
+        return CaseStructure(
+            attrs={
+                'create': True,
+                'case_type': 'episode',
+                'update': {
+                    'treatment_supporter_mobile_number': outcome.PatientId.cmob,
+                },
+            },
+            indices=[CaseIndex(
+                self.occurrence(outcome),
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=self.occurrence(outcome).attrs['case_type'],
+            )],
+        )
+
+    def test(self, followup):
+        episode_structure = self.episode(
+            Outcome.objects.get(PatientId=followup.PatientID)
+        )
+        return CaseStructure(
+            attrs={
+                'create': True,
+                'case_type': 'test',
+                'update': {
+                    'date_tested': followup.TestDate,
+                },
+            },
+            indices=[CaseIndex(
+                episode_structure,
+                identifier='host',
+                relationship=CASE_INDEX_EXTENSION,
+                related_type=episode_structure.attrs['case_type'],
+            )],
+        )
+
+    @property
+    @memoized
+    def _outcomes(self):
+        return Outcome.objects.filter(PatientId=self.patient_detail)
+
+    @property
+    @memoized
+    def _followups(self):
+        return Followup.objects.filter(PatientID=self.patient_detail)
+
+    @property
+    def _location(self):
+        return self.nikshay_code_to_location[self._nikshay_code]
+
+    @property
+    def _nikshay_code(self):
+        return '-'.join(self.patient_detail.PregId.split('-')[:4])
+
+
+class Command(BaseCommand):
+
+    def handle(self, *args, **options):
```

Comment:
can you think about / implement what would need to change if the migration should update existing data? I have the feeling we're going to want that to work and it might inform the architecture a little bit (e.g. I don't think you can use the `CaseStructure` objects exactly how you're using them since they won't always create the cases...)